### PR TITLE
Add ARM support

### PR DESCRIPTION
**Pull request by sylveon**
Issue: None

## PR Type
What kind of change does this PR introduce?
- Feature
- Build or CI related changes

## What is the current behavior?
An app targeting ARM (not ARM64) cannot be built using Microsoft.Toolkit.Win32.UI.XamlHost because it does not have ARM binaries.

## What is the new behavior?
ARM binaries are built and added to the NuGet package.

## PR Checklist
- [x] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] Pull Request has been submitted to the documentation repository. Link: Nothing to adjust in current docs
- [x] Contains **NO** breaking changes

### CODE REVIEW

**Comment by azchohfi** [comment]
@ocalvo does this makes sense even though we don't support ARM32 in WPF and WinForms? A simple C++ Win32 app can still use this, right? I believe it does make sense. Why did we opt not to add ARM32 binaries? Microsoft.Windows.CppWinRT supports ARM32.

**Comment by sylveon** [comment]
Thanks, hadn't noticed that. Added.

File: Microsoft.Toolkit.Win32.UI.XamlApplication.Package/Microsoft.Toolkit.Win32.UI.XamlApplication.Package.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <Description>This library provides XAML islands common helpers for native applications. It is part of the Windows Community Toolkit.</Description>
     <PackageTags>XAML Islands XAMLHost</PackageTags>
     <PackageId>Microsoft.Toolkit.Win32.UI.XamlApplication</PackageId>
-    <Platforms>x64;x86;ARM64;AnyCPU</Platforms>
+    <Platforms>x64;x86;ARM;ARM64;AnyCPU</Platforms>
```
Comment (ocalvo) [inline Microsoft.Toolkit.Win32.UI.XamlApplication.Package/Microsoft.Toolkit.Win32.UI.XamlApplication.Package.csproj]: This should be only AnyCPU as this is a NuGet package only.

File: Microsoft.Toolkit.Win32.UI.XamlApplication.Package/Microsoft.Toolkit.Win32.UI.XamlApplication.Package.csproj [hunk 2]
```diff
@@ -61,6 +61,23 @@
       <PackagePath>runtimes\win10-x64\native</PackagePath>
     </Content>
 
+    <Content Include="..\arm\$(Configuration)\Microsoft.Toolkit.Win32.UI.XamlApplication\Microsoft.Toolkit.Win32.UI.XamlHost.dll">
+      <Pack>true</Pack>
+      <PackagePath>runtimes\win10-arm\native</PackagePath>
+    </Content>
+
+    <!--
```
Comment (ocalvo) [inline Microsoft.Toolkit.Win32.UI.XamlApplication.Package/Microsoft.Toolkit.Win32.UI.XamlApplication.Package.csproj]: I believe the PDB issue has been resolved as we are flattening WAP projects, so this no longer required and we can safely ship the PDBs

Comment (azchohfi) [reply Microsoft.Toolkit.Win32.UI.XamlApplication.Package/Microsoft.Toolkit.Win32.UI.XamlApplication.Package.csproj]: Let's fix this on another PR.

File: Microsoft.Toolkit.Win32.UI.XamlApplication.Package/Microsoft.Toolkit.Win32.UI.XamlApplication.Package.csproj (hunk 1 above)
Comment (azchohfi) [reply Microsoft.Toolkit.Win32.UI.XamlApplication.Package/Microsoft.Toolkit.Win32.UI.XamlApplication.Package.csproj]: Let's fix this on another PR.

**Comment by azchohfi** [comment]
Let's fix this on another PR.

**Comment by azchohfi** [comment]
Thanks @sylveon!

File: Microsoft.Toolkit.Win32.UI.XamlApplication.Package/Microsoft.Toolkit.Win32.UI.XamlApplication.Package.csproj (hunk 2 above)
Comment (sylveon) [reply Microsoft.Toolkit.Win32.UI.XamlApplication.Package/Microsoft.Toolkit.Win32.UI.XamlApplication.Package.csproj]: I'm curious about what this means exactly. Are all WAP projects now flattened, meaning the previous workaround of adding the following target to the `.wapproj` is now not needed anymore?
```xml
<Target Name="FlattenPackage" BeforeTargets="_ConvertItems">
  <ItemGroup>
    <_TemporaryFilteredWapProjOutput Include="@(_FilteredNonWapProjProjectOutput)" />
    <_FilteredNonWapProjProjectOutput Remove="@(_TemporaryFilteredWapProjOutput)" />
    <_FilteredNonWapProjProjectOutput Include="@(_TemporaryFilteredWapProjOutput)">
      <SourceProject></SourceProject>
    </_FilteredNonWapProjProjectOutput>
  </ItemGroup>
</Target>
```

**Pull request merged**